samples/sensor/lis2dw12/src/main.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2024 Pierrick Curt
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+
+static void fetch_and_display(const struct device *sensor)
+{
+	static unsigned int count;
+	struct sensor_value accel[3];
+	struct sensor_value temperature;
+	int rc;
+
+	rc = sensor_sample_fetch_chan(sensor, SENSOR_CHAN_ACCEL_XYZ);
+	if (rc < 0) {
+		printf("ERROR: SENSOR_CHAN_ACCEL_XYZ fetch failed: %d\n", rc);
+	}
+	rc = sensor_sample_fetch_chan(sensor, SENSOR_CHAN_DIE_TEMP);
+	if (rc < 0) {
+		printf("ERROR: SENSOR_CHAN_DIE_TEMP fetch failed: %d\n", rc);
+	}
+
+	++count;
+	if (rc == 0) {
+		rc = sensor_channel_get(sensor, SENSOR_CHAN_ACCEL_XYZ, accel);
+		if (rc == 0) {
+			rc = sensor_channel_get(sensor, SENSOR_CHAN_DIE_TEMP, &temperature);
+		}
+	}
+	if (rc < 0) {
+		printf("ERROR: Update failed: %d\n", rc);
+	} else {
+		printf("#%u @ %u ms: x %f , y %f , z %f, t %f\n", count, k_uptime_get_32(),
+		       sensor_value_to_double(&accel[0]), sensor_value_to_double(&accel[1]),
+		       sensor_value_to_double(&accel[2]), sensor_value_to_double(&temperature));
+	}
+}
+
+#ifdef CONFIG_LIS2DW12_TRIGGER
+
+static void lis2dw12_trigger_handler(const struct device *dev, const struct sensor_trigger *trig)
+{
+	fetch_and_display(dev);
+}
+#endif
+
+int main(void)
+{
+	const struct device *const sensor = DEVICE_DT_GET_ANY(st_lis2dw12);
+
+	if (sensor == NULL) {
+		printf("No device found\n");
+		return 0;
+	}
+	if (!device_is_ready(sensor)) {
+		printf("Device %s is not ready\n", sensor->name);
+		return 0;
+	}
+
+#ifdef CONFIG_LIS2DW12_TRIGGER
+	struct sensor_trigger trig;
+
+	trig.type = SENSOR_TRIG_DOUBLE_TAP;
+	trig.chan = SENSOR_CHAN_ACCEL_XYZ;
+	sensor_trigger_set(sensor, &trig, lis2dw12_trigger_handler);
+
+	while (true) {
+		k_sleep(K_MSEC(2000));
+	}
+
+#else  /* CONFIG_LIS2DW12_TRIGGER */
+	printf("Polling at 0.5 Hz\n");
+	while (true) {
+		fetch_and_display(sensor);
+		k_sleep(K_MSEC(2000));
+	}
+#endif /* CONFIG_LIS2DW12_TRIGGER */
```

Comment:
I would keep a single ```while(true)``` block, with ```fetch_and_display()``` inside a #ifdef. The configuration of trigger under another ifdef above the while block.

Reply:
If I do this I need two `#ifdef CONFIG_LIS2DW12_TRIGGER` conditions:
- one around the sensor_trigger_set bloc
- one around the` fetch_and_display(sensor)`
To keep a good code readability I prefer keep only one.

What do you think ?

Reply:
Well, I think that maybe at the end it's just a matter of preference. I've nothing against it, just a matter of preference.